python/ucxx/ucxx/utils.py

```diff
@@ -150,16 +153,31 @@ def get_address(ifname=None):
 
     >>> get_address(ifname='lo')
     '127.0.0.1'
+
+    >>> get_address(ifname='lo', use_ipv6=True)
+    '::1'
     """
 
     def _get_address(ifname):
-        ifname = ifname.encode()
-        with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
-            return socket.inet_ntoa(
-                fcntl.ioctl(
-                    s.fileno(), 0x8915, struct.pack("256s", ifname[:15])  # SIOCGIFADDR
-                )[20:24]
-            )
+        if use_ipv6:
+            import psutil
+
+            addrs = psutil.net_if_addrs()
+            if ifname in addrs:
+                for addr in addrs[ifname]:
+                    if addr.family == socket.AF_INET6:
+                        return addr.address.split("%")[0]
+            return None
```

Comment:
Raise an exception here? Or add `None` to the docstring

Reply:
Good catch, fixed in https://github.com/rapidsai/ucxx/pull/387/commits/463b5f94070fffd6ba41c3633a8b11243de2f667 .